lib/country_lookup.rb

```diff
@@ -99,20 +99,35 @@ def initialize_from_file
   end
 
   def self.initialize
+    if !@initialize_bg_thread.nil?
+      @initialize_bg_thread.kill.join
+    end
     @lookup = Lookup.new
     return nil
   end
 
+  def self.initialize_async
+    if !@initialize_bg_thread.nil? && @initialize_bg_thread.alive?
+      @initialize_bg_thread.kill.join
+    end
+    @initialize_bg_thread = Thread.new { @lookup = Lookup.new }
+    return @initialize_bg_thread
+  end
+
+  def self.is_ready_for_lookup
+    !@lookup.nil?
+  end
+
   def self.lookup_ip_string(ip_string)
     if @lookup.nil?
-      @lookup = Lookup.new
+      initialize
```

Comment:
So this is still going to call initialize on the main thread if it hasnt finished and this gets used, right?  Still dont think this is right

Reply:
We could wait on the original thread instead. Or were u proposing something else? Like return a custom eval reason until it's initialized